Tests/SystemTests/ChannelModesTests.swift

```diff
@@ -0,0 +1,179 @@
+            resolutionPolicyFactory: MockResolutionPolicyFactory(),
+            ablyService: DefaultAblyPublisherService(configuration: publisherConnectionConfiguration),
+            locationService: defaultLocationService,
+            routeProvider: MockRouteProvider()
+        )
+        
+    }
+    
+    private func createSubscriber(trackableId: String, clientId: String, ablyApiKey: String) -> AblyAssetTrackingSubscriber.Subscriber {
+        let subscriberConnectionConfiguration = ConnectionConfiguration(apiKey: ablyApiKey, clientId: clientId)
+        let resolution = Resolution(accuracy: .balanced, desiredInterval: 500, minimumDisplacement: 100)
+        
+        return SubscriberFactory.subscribers()
+            .connection(subscriberConnectionConfiguration)
+            .resolution(resolution)
+            .log(LogConfiguration())
+            .delegate(self)
+            .trackingId(trackableId)
+            .start(completion: { _ in })!
+    }
+    
+    private func getChannelMetrics(trackableId: String, ablyApiKey: String) throws -> ChannelMetrics? {
+        let apiKeyParts = ablyApiKey.split(separator: ":")
+        
+        guard apiKeyParts.count == 2 else {
+            XCTFail("Invalid number of api key parts. Expected 2, got \(apiKeyParts.count)")
+            return nil
+        }
+        
+        let auth = try XCTUnwrap((apiKeyParts[0] + ":" + apiKeyParts[1]).data(using: .utf8)?.base64EncodedString())
+        let credentials = "Basic \(auth)"
+        let url = try XCTUnwrap(URL(string: "https://rest.ably.io/channels/tracking:\(trackableId)"))
+        
+        var request = URLRequest(url: url)
+        request.addValue(credentials, forHTTPHeaderField: "Authorization")
+        
+        let didFinishDataTaskExpectation = self.expectation(description: "Get Channel Metrics Request - did finished")
+        var channelMetricsObject: ChannelMetrics?
+        
+        let task = URLSession.shared.dataTask(with: request) { data, _, error in
+            if let error = error {
+                XCTFail("Get Channel Metrics Request Failed with error: \(error)")
+                didFinishDataTaskExpectation.fulfill()
+                return
+            }
+            
+            guard let data = data else {
+                XCTFail("Get Channel Metrics Request `data` is `nil`")
+                didFinishDataTaskExpectation.fulfill()
+                return
+            }
+                        
+            do {
+                channelMetricsObject = try JSONDecoder().decode(ChannelMetrics.self, from: data)
+            } catch {
+                XCTFail("Get Channel Metrics Request - JSON Decode failed with error: \(error)")
+            }
+            
+            didFinishDataTaskExpectation.fulfill()
+        }
+    
+        task.resume()
+        
+        wait(for: [didFinishDataTaskExpectation], timeout: defaultDelayTime)
+        
+        return channelMetricsObject
+    }
+    
+
+    
```

Comment:
Is this amount of space intentional?

Reply:
removed here a01abbc9daa67c842a9af2d3dcc0e2ab882f7a06